scripts/parse_status.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
+import re
+import sys
+
+# Parses a common style of defining enums in c, printing each with their hex value.
+# With small modifications, this could be used to generate a function to convert status codes to strings.
+# 
+# To parse the statuses in this repo
+# cat src/include/com/amazonaws/kinesis/video/webrtcclient/Include.h | ./scripts/parse_status.py 
+
+'''
+# Example usage (uncomment as a basic test)
+paragraph="""
+#define STATUS_PARENT 0x4
+#define STATUS_CHILD STATUS_PARENT + 0x1
+"""
+operands_map = operands_by_name(paragraph) 
+print(operands_map) # {'STATUS_CHILD': ('STATUS_PARENT', 1), 'STATUS_PARENT': (None, 4)}
+example_sums = hex_sums(operands_map)
+print(example_sums) # {'STATUS_CHILD': "0x5", 'STATUS_PARENT': "0x4"}
+'''
+
+pattern = re.compile("#define *([A-Z_]*) *(([A-Z_]*) *\+ *)?0x([0-9]*)")
+
+def operands_by_name(paragraph):
+    matches = filter(None, [pattern.match(line) for line in paragraph.splitlines()])
+    return {groups[0]: (groups[2], int(groups[3], base=16)) for groups in 
+            [match.groups() for match in matches]}
+
+def sum_value(by_name, name):
+    base, idx = by_name[name]
+    return idx if base is None else idx + sum_value(by_name, base)
+
+def hex_sums(by_name):
+    return {name: hex(sum_value(operands_map, name)) for name in by_name.keys()}
+
+paragraph = sys.stdin.read()
+operands_map = operands_by_name(paragraph)
+sums_map = hex_sums(operands_map)
+lines = ["{} {}".format(name, value) for name, value in sums_map.items()]
```

Comment:
It would be nice if the status codes could be aligned as well so they're easier to see, but not necessary.

Reply:
Good idea, done:

```
...
STATUS_SIGNALING_NO_PAYLOAD_IN_MESSAGE                             0x5d000002
STATUS_SIGNALING_DUPLICATE_MESSAGE_BEING_SENT                      0x5d000002
STATUS_SIGNALING_ICE_TTL_LESS_THAN_GRACE_PERIOD                    0x5d000002
STATUS_SIGNALING_DISCONNECTED_CALLBACK_FAILED                      0x5d000002
STATUS_SIGNALING_INVALID_MESSAGE_TTL_VALUE                         0x5d000002
STATUS_SIGNALING_ICE_CONFIG_REFRESH_FAILED                         0x5d000002
STATUS_SIGNALING_RECONNECT_FAILED                                  0x5d000030
STATUS_SIGNALING_DELETE_CALL_FAILED                                0x5d000031
...
```